classes/dev/ComposerScript.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * @file classes/dev/ComposerScript.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ComposerScript
+ *
+ * @brief Custom composer scripts to run post installs/updates
+ */
+
+namespace PKP\dev;
+
+use Exception;
+
+require_once '../../tools/bootstrap.inc.php';
+import('lib.pkp.classes.file.FileManager');
+
+class ComposerScript
+{
+	/**
+	 * A post-install-cmd custom composer script that
+	 * copies composer installs from repositories
+	 * to the correct/existing directories of the following dependencies:
+	 * jquery-ui and jquery validation
+	 */
+	public static function copyVendorAssets(): void
+	{
+		$fileManager = new \FileManager();
+		$vendorBaseDir = __DIR__ . '/../../lib/vendor';
+		$jsPluginsDir = __DIR__ . '/../../js/lib';
+
+		$source = [
+			'jquery-ui.js' => $vendorBaseDir . '/jquery/ui/dist/jquery-ui.js',
+			'jquery-ui.min.js' => $vendorBaseDir . '/jquery/ui/dist/jquery-ui.min.js',
+			'jquery-validate' => $vendorBaseDir . '/jquery/validation/dist'
+		];
+
+		$dest = [
+			'jquery-ui.js' => $vendorBaseDir . '/components/jqueryui/jquery-ui.js',
+			'jquery-ui.min.js' => $vendorBaseDir . '/components/jqueryui/jquery-ui.min.js',
+			'jquery-validate' => $jsPluginsDir . '/jquery/plugins/validate'
+		];
+
+		// jQuery UI
+		$fileManager->copyFile($source['jquery-ui.js'], $dest['jquery-ui.js']) || throw new Exception('Failed to copy jquery-ui.js to destination folder');
```

Comment:
Just tagging @asmecher to confirm if it's ok this breaking change, regarding the required permissions (https://docs.pkp.sfu.ca/admin-guide/en/troubleshooting#configuring-file-permissions).

Reply:
This script is run by Composer during dependency installation, not when OJS is installed and in production. For the most part, it'll be run from the package build script when I build a release, and for folks that host OJS from git it'll be run early in the installation process from the same terminal that's used to e.g. clone the code into place. I don't think the production file permission document is relevant. (This is also why I [recommended not using `FileManager`](https://github.com/pkp/pkp-lib/pull/10167#discussion_r1722204962) -- it's intended for execution in the bootstrapped PKP environment, not elsewhere.)